mitmproxy/addons/next_layer.py

```diff
@@ -167,14 +167,15 @@ def s(*layers):
 
         # 5)  Handle application protocol
         # 5a) Is it DNS?
-        if udp_based:
-            try:
-                # TODO: DNS over TCP
-                dns.Message.unpack(data_client)  # TODO: perf
-            except struct.error:
-                pass
-            else:
-                return layers.DNSLayer(context)
+        is_dns = (
+            udp_based
+            and starts_like_dns_record(data_client)
+            or tcp_based
+            and starts_like_dns_over_tcp_record(data_client)
+        )
+        if is_dns:
+            return layers.DNSLayer(context)
```

Comment:
I think we should just set is_dns based on the destination port (53 or 5353) for now and remove the heuristic methods. `starts_like_dns_over_tcp_record` will currently return False if the first packet is not complete yet, which can happen for DNS over TCP. 53/5353 should be good enough really since you cannot configure the DNS port typically.

Reply:
the following test would fail if we were to use this heuristic:
https://github.com/mitmproxy/mitmproxy/blob/fc24f85929db05b615f5fb6f028a10388dd54de9/test/mitmproxy/addons/test_next_layer.py#L673-L682
how would we want to handle this?

Reply:
Let's just change the port for the test. If we find that raw traffic on 53 is a problem (or DNS on other ports), we can come back to this and tune it later. My gut feeling is that we should be fine.